src/java.base/share/classes/java/nio/Heap-X-Buffer.java.template

```diff
@@ -290,7 +290,8 @@ class Heap$Type$Buffer$RW$
     // or the full sequence of chars is being appended, copying the chars to
     // an intermedite String in StringBuilder::toString is avoided.
     //
-    private $Type$Buffer appendChars(CharSequence csq, int start, int end) {
+    public $Type$Buffer append(CharSequence csq, int start, int end) {
+#if[rw]
         checkSession();
 
         Objects.checkFromToIndex(start, end, csq.length());
```

Comment:
The append(CharSequence) methods are specified to append "null" when the CharSequence is null so append(null, 0, 0) will need to be handled here.

Reply:
Based on code inspection, it looks like that error might have already been possible but did not turn up before.